### PR TITLE
Allow creating multiple AWS integrations

### PR DESCRIPTION
**Pull request by cw75**
## Describe your changes and why you are making these changes
This PR enables users to create multiple AWS integrations. Each integration has its own Terraform directory that manages a cluster.

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/cmd/server/server/execute_handler.go [hunk 1]
```diff
@@ -25,15 +25,7 @@ func HandleError(
...
-	var externalMsg string
-	dbxErr, ok := err.(errors.DropboxError)
-	if ok {
-		// Only return the top-level error message to clients.
-		externalMsg = dbxErr.GetMessage()
-	} else {
-		externalMsg = err.Error()
-	}
-	response.SendErrorResponse(w, externalMsg, statusCode)
+	response.SendErrorResponse(w, errors.GetMessage(err), statusCode)
```
Comment (cw75) [inline src/golang/cmd/server/server/execute_handler.go:28]: @kenxu95 I found that `errors.GetMessage(err)` has the same functionality as the original code, with the added benefit that it gets errors at all levels instead of just the top-level, which I found to be much more useful both to ourselves and to the users.

File: sdk/aqueduct/client.py [hunk 2]
```diff
@@ -427,7 +427,7 @@ def integration(
                 metadata=integration_info,
             )
         elif integration_info.service == ServiceType.AWS:
-            dynamic_k8s_integration_name = "%s:k8s" % name
+            dynamic_k8s_integration_name = "%s:aqueduct_ondemand_k8s" % name
```
Comment (kenxu95) [inline sdk/aqueduct/client.py:430]: Curious why this change?

File: src/golang/cmd/executor/executor/dynamic_teardown.go [hunk 3]
```diff
@@ -31,52 +33,62 @@ func (ex *DynamicTeardownExecutor) Run(ctx context.Context) error {
...
+						dynamicIntegration,
+						ex.IntegrationRepo,
+						ex.Database,
+					); err != nil {
+						log.Error(errors.Wrap(err, "Unable to delete dynamic k8s integration"))
+						return
+					}
+				} else {
+					log.Info("Have not reached keepalive threshold, not tearing down the cluster.")
+				}
```
Comment (kenxu95) [inline src/golang/cmd/executor/executor/dynamic_teardown.go:87]: Now that it's a goroutine, should we have some timeout mechanism? Actually, was there a timeout mechanism before, or do our executors just run unbounded.

File: src/golang/cmd/server/handler/cloud_integration_utils.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	"github.com/aqueducthq/aqueduct/lib/dynamic"
+	"github.com/aqueducthq/aqueduct/lib/errors"
+	"github.com/aqueducthq/aqueduct/lib/lib_utils"
+	"github.com/aqueducthq/aqueduct/lib/models/shared"
+	"github.com/aqueducthq/aqueduct/lib/workflow/operator/connector/auth"
+	"github.com/google/uuid"
+)
+
+// setupCloudIntegration sets up the cloud integration's Terraform directory, registers a k8s
+// integration and run `terraform init` to initialize the Terraform workspace.
```
Comment (kenxu95) [inline src/golang/cmd/server/handler/cloud_integration_utils.go:21]: Can you add a note here about where this terraform directory exists?

File: src/golang/cmd/server/server/execute_handler.go (hunk 1 above)
Comment (kenxu95) [reply src/golang/cmd/server/server/execute_handler.go:28]: Actually since this is such a useful and scoped changed, can you make a new PR with just this? I can stamp it immediately (might want it for my error handling work)

File: src/golang/lib/dynamic/engine.go [hunk 5]
```diff
@@ -30,9 +32,12 @@ const (
 	K8sClusterUpdateAction k8sClusterActionType = "update"
 )
 
-const stateLockErrMsg = "Error acquiring the state lock"
+const (
+	stateLockErrMsg      = "Error acquiring the state lock"
+	K8sClusterNameSuffix = "aqueduct_ondemand_k8s"
```
Comment (kenxu95) [inline src/golang/lib/dynamic/engine.go:37]: This is the name of the K8s integration right? Not the eks cluster name.

**Comment by kenxu95** [comment]
Looks good but left a couple questions!

File: sdk/aqueduct/client.py (hunk 2 above)
Comment (cw75) [reply sdk/aqueduct/client.py:430]: just to prevent this name from colliding with user-defined names. `aqueduct_ondemand_k8s` is much less likely to collide than `k8s`

File: src/golang/cmd/executor/executor/dynamic_teardown.go (hunk 3 above)
Comment (cw75) [reply src/golang/cmd/executor/executor/dynamic_teardown.go:87]: Our executor runs unbounded, but in this case Terraform itself has timeout built in so we won't run forever even during the error case.

File: src/golang/cmd/server/handler/cloud_integration_utils.go (hunk 4 above)
Comment (cw75) [reply src/golang/cmd/server/handler/cloud_integration_utils.go:21]: added a note to `setupTerraformDirectory`

File: src/golang/lib/dynamic/engine.go (hunk 5 above)
Comment (cw75) [reply src/golang/lib/dynamic/engine.go:37]: yeah, updated to `K8sIntegrationNameSuffix`

**Comment by cw75** [comment]
@kenxu95 addressed all comments.

File: src/golang/cmd/executor/executor/dynamic_teardown.go (hunk 3 above)
Comment (kenxu95) [reply src/golang/cmd/executor/executor/dynamic_teardown.go:87]: got it. Can you note that at the beginning of the goroutine? It's one of the first things that comes to my mind when reading unbounded goroutines (usually I've been taught to use ErrGroups where all goroutines share the same context).

**Review by kenxu95 — APPROVED** [review]
